k2/csrc/tensor_ops.h

```diff
@@ -106,6 +109,26 @@ Tensor IndexAdd(Tensor &src, Array1<int32_t> &indexes, int32_t dim,
 void IndexAdd(Tensor &src, Array1<int32_t> &indexes, bool allow_minus_one,
               Tensor *dest);
 
+/*
+  Returns a 1-D Tensor that is a result of indexing 1-D `src` with Ragged array
+  `indexes` whose NumAxes() is 2. ans.Dims()[0] will equal to indexes.Dims0() as
+  we suppose there is at most one non-zero element in `src` for any indexes
+  sub-list in `indexes`.
+
+     @param [in] src  Source 1-D tensor, to be indexed.
+     @param [in] indexes   Indexes to use whose NumAxes() == 2, for any
+                      sub-list `i` in `indexes`, we suppose there is at most
+                      one non-zero element in `src` and we'll set ans[i]
+                      with that non-zero element; if all elements for
```

Comment:
Please be more precise...  explain what we will set ans[i] to, e.g. is it src[that_element] or that_element?

Reply:
... also (this is probably my fault for suggesting this interface, but...) it might be more elegant to create a function that would create an `Array1<int32_t>` containing (the nonzero element) if it exists, else -1; and then use our normal IndexSelect() to index with that array.  But this interface is OK too I guess.

Reply:
I suupose you mean return the indexes (or -1) of the nonzero values... will change to that, thanks!

Reply:
wait.. I think I realized what happened here.
This was supposed to be  an optimziation.

Reply:
So don't change the code.. just clarify the documentation.

Reply:
OK, then the current PR would be ready to merge with my last commit, noted I changed `elements` in documentation to `values in src`, so there should be no misunderstanding